### PR TITLE
Feature/ts type 2.0 transition

### PR DESCRIPTION
**Pull request by W3stside**
1. Transition `DutchAuction` (`DutchExchange.sol` interface) into 2.0

### CODE REVIEW

File: src/api/types.ts [hunk 1]
```diff
@@ -81,44 +81,77 @@ export interface Receipt {
...
+    auctioneer: Account,
+    ethUSDOracle: Account,
+    thresholdNewTokenPair: Balance,
+    thresholdNewAuction: Balance,
+    tx?: TransactionObject,
+  ): Promise<Receipt>,
   updateApprovalOfToken(token: Account, approved: boolean, tx?: TransactionObject): Promise<Receipt>,
-  updateETHUSDPriceOracle(_ETHUSDOracletoken: Account, tx?: TransactionObject): Promise<Receipt>,
+  startMasterCopyCountdown(masterCopy: Account, tx?: TransactionObject): Promise<Receipt>,
+  updateMasterCopy(tx?: TransactionObject): Promise<Receipt>,
```
Comment (Velenir) [inline src/api/types.ts]: For some of these `tx` parameter isn't optional as `msg.sender` is checked inside the function.

Comment (W3stside) [reply src/api/types.ts]: hmm thought I had changed that. Thanks

File: src/api/types.ts [hunk 2]
```diff
@@ -155,22 +186,64 @@ export interface DXAuction {
...
+  computeRatioOfHistoricalPriceOraclesForJS(
+    tokenA: Account,
+    tokenB: Account,
+    auctionIndex: Index,
+  ): Promise<[BigNumber, BigNumber]>,
+
+  // internal
+  getTokenOrder(tokenA: Account, tokenB: Account): never,
+  setAuctionStart(tokenA: Account, tokenB: Account, value: number): never,
+  resetAuctionStart(tokenA: Account, tokenB: Account): never,
```
Comment (Velenir) [inline src/api/types.ts]: `internal` methods are not exposed on the contract in JS, no need to add them

**Comment by W3stside** [comment]
Failing Travis checks that are being fixed in incoming PR

**Review by Velenir — APPROVED** [review]

**Pull request merged**